lib/awful/layout/suit/tile.lua.in

```diff
@@ -156,26 +155,28 @@ local function tile_group(cls, wa, orientation, fact, group)
     size = math.min(size, available)
 
     local coord = wa[y]
-    local geom = {}
     local used_size = 0
     local unused = wa[height]
     for c = group.first,group.last do
         local i = c - group.first +1
-        geom[width] = size - cls[c].border_width * 2
-        geom[height] = math.floor(unused * fact[i] / total_fact) - cls[c].border_width * 2
+        local geom = {}
+        geom[width] = size
+        geom[height] = math.floor(unused * fact[i] / total_fact)
         geom[x] = group.coord
         geom[y] = coord
-        geom = cls[c]:geometry(geom)
```

Comment:
I'm fine with this PR just being about "make layout return info instead of applying directly". IMO that produces some nicer API. We'd just need not to have a regression with this, so `c:apply_size_hints` is needed.

For the useless gap: I am under the impression that "everyone" has awesome ignore size hints anyway, we just have to have the code for standard compliance.

I'm fine with first applying useless gaps and then size hints. This should give you the biggest size that's compatible with the hints and the gaps. Sure, the gaps will be "unevenly large". Everyone who dislikes this can ignore size hints.